### DISCUSSION TITLE
Edge gets reinstalled when running Sophia.

### DISCUSSION THREAD

**Discussion by fabcard** [farag2/Sophia-Script-for-Windows#541] [General]
Hello.
Could you please stop edge reinstalling every time I run Sophia? Or maybe add an option on script where we can choose to install edge or not. It's really annoying to have to uninstall it always. I don't use edge and my system runs fine without it.
Thanks in advance.

**Comment by farag2** [comment]
Hi. :) Sophia was made to heal Windows after malicious tweakers which uninstall system components like MS Edge. So Sophia Script will always install it if detects that it was uninstalled by some reasons.

Moral: do not break Windows by uninstalling system components even if you think that it Edge installed has any effect on system speed. This is a lie told by many so called bloggers that has any effect on Windows. It mustn't be uninstalled being necessary app. Hope, I explained my position. :)

**Comment by fabcard** [comment]
Hi @farag2 
Excuse my poor English, it's not my main language. As I said before, I don't use edge and my system works fine without it. I'm not uninstalling it because I think it slows down my computer, I have other reasons for that. Also, if the edge uninstallers are malicious or not, it's my opinion and responsibility on what I want or not on my system. Your Sophia script already has many options to remove or changes apps on Windows, so in my opinion, adding or removing edge should be an option to choose as well. Is there any chance for you to add a way to bypass the edge reinstallation every time I run the script, please?
Again, thank you.

**Comment by farag2** [comment]
> I don't use edge and my system works fine without it.

You may not using it but this is a system component and you mustn't uninstall regardless you think your system stability is OK (not) or not.

> I think it slows down my computer

How a built-in browser may slow your computer? Who told you that, another blogger from YouTube who counts FPS in a fake video?

> it's my opinion and responsibility on what I want or not on my system

Fully agree. But Sophia Script will always restore system components if it's possible.

> adding or removing edge should be an option to choose as well.

Won't happen...

> Is there any chance for you to add a way to bypass the edge reinstallation every time I run the script, please?\

Better remove that block from the code for yourself. The code is open — anyone can modify it for their needs.
Remove this check: 

https://github.com/farag2/Sophia-Script-for-Windows/blob/048c472448610f499f592b5f1d30429685336bba/src/Sophia_Script_for_Windows_10/Module/Sophia.psm1#L92
